src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const splitString = sourceString
+    .split(';')
+    .map((item) => item.split(':'))
+    .map((item) => item.map((value) => value.trim()))
+    .map((item) => item.filter((value) => value !== ''))
+    .filter((item) => item.length !== 0)
+    .reduce((object, item) => {
+      object[item[0]] = item[1];
+
+      return object;
+    }, {});
+
+  return splitString;
```

Comment:
The variable `splitString` is being returned, which is actually an object. Consider renaming it to something more descriptive like `resultObject` to better reflect its purpose.